src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoTreeNodes.cs

```diff
@@ -681,65 +694,164 @@ internal partial class DatabaseChildFactory : SmoChildFactoryBase
 
         protected override void OnExpandPopulateFolders(IList<TreeNode> currentChildren, TreeNode parent)
         {
+			if (WorkspaceService<SqlToolsSettings>.Instance.CurrentSettings.SqlTools.ObjectExplorer.SchemaBasedOE)
+			{
+                currentChildren.Add(new FolderNode {
+                    NodeValue = SR.SchemaHierarchy_Schemas,
+                    NodeType = "Folder",
+                    NodeTypeId = NodeTypes.ExpandableSchemas,
+                    IsSystemObject = false,
+                    SortPriority = SmoTreeNode.NextSortPriority,
+                });
+			}
+			if (!WorkspaceService<SqlToolsSettings>.Instance.CurrentSettings.SqlTools.ObjectExplorer.SchemaBasedOE)
+			{
+                currentChildren.Add(new FolderNode {
+                    NodeValue = SR.SchemaHierarchy_Tables,
+                    NodeType = "Folder",
+                    NodeTypeId = NodeTypes.Tables,
+                    IsSystemObject = false,
+                    SortPriority = SmoTreeNode.NextSortPriority,
+                });
+			}
+			if (!WorkspaceService<SqlToolsSettings>.Instance.CurrentSettings.SqlTools.ObjectExplorer.SchemaBasedOE)
+			{
+                currentChildren.Add(new FolderNode {
+                    NodeValue = SR.SchemaHierarchy_Views,
+                    NodeType = "Folder",
+                    NodeTypeId = NodeTypes.Views,
+                    IsSystemObject = false,
+                    SortPriority = SmoTreeNode.NextSortPriority,
+                });
+			}
+			if (!WorkspaceService<SqlToolsSettings>.Instance.CurrentSettings.SqlTools.ObjectExplorer.SchemaBasedOE)
+			{
+                currentChildren.Add(new FolderNode {
```

Comment:
Is there a usecase where a `FolderNode`'s `NodeType` will not be `Folder`?
Feels like something that can be taken care by `FolderNode` constructor, and could be avoided when creating these objects.

Can be followed up later.